dfe-analytics.gemspec

```diff
@@ -36,6 +36,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rubocop-rspec', '~> 2'
   spec.add_development_dependency 'solargraph'
   spec.add_development_dependency 'sqlite3'
+  spec.add_development_dependency 'store_model'
```

Comment:
Do we have to have this whole thing? Isn't the point the interface — ie that SomeObject implements to_json?

Reply:
eg

```ruby
ObjectWithToJson = Struct.new(:colour) do
   def to_json
     JSON.generate({colour: colour})
   end
end
```

or something?

Reply:
It's not quite `to_json`, I changed the implementation to check for `as_json`. We could hand-roll an `as_json` but I thought it would be better test to just use pre-built library. Happy to discuss.

Reply:
```ruby
ObjectWithAsJson = Struct.new(:colour) do
  def as_json
    {colour: colour}
  end
end
```

?

Reply:
^ imo this says "your object should support `as_json`", using the gem says "we support this particular gem", and I think I prefer the former?

Reply:
Considering the spec is failing in CI because of this ... maybe hand-rolling will actually end up just being easier. _sigh_

Reply:
I've adopted the Struct approach, thanks @duncanjbrown !